testing/src/test/scala/org/http4s/testing/ErrorReporting.scala

```diff
@@ -9,11 +9,81 @@
 
 package org.http4s.testing
 import java.io.{ByteArrayOutputStream, PrintStream}
-
+import cats.implicits._
+import cats.Monad
+import org.http4s.{Headers, MessageFailure, Request, Response, Status}
+import org.http4s.headers.{Connection, `Content-Length`}
+import org.http4s.syntax.string._
 import scala.util.control.NonFatal
+import java.net.URL
+import java.security.{
+  AccessControlContext,
+  CodeSource,
+  Permissions,
+  PrivilegedAction,
+  ProtectionDomain
+}
 
 object ErrorReporting {
 
+  /**
+    * Silences System.out and System.err streams for the duration of thunk.
+    * Restores back the original streams before exiting.
+    */
+  def silenceOutputStreams[R](thunk: => R): R =
+    synchronized {
+      val originalOut = System.out
+      val originalErr = System.err
+
+      // Create blank permissions
+      val perm = new Permissions
+
+      // CodeSource domain for which these permissions apply (all files)
+      val certs: Array[java.security.cert.Certificate] = null
+      val code: CodeSource = new CodeSource(new URL("file:/*"), certs)
+
+      val domain = new ProtectionDomain(code, perm)
+      val domains = new Array[ProtectionDomain](1)
+      domains(0) = domain
+      val context = new AccessControlContext(domains)
+
+      // Redirect output to dummy stream
+      val fakeOutStream = new PrintStream(new ByteArrayOutputStream)
+      val fakeErrStream = new PrintStream(new ByteArrayOutputStream)
```

Comment:
This is good if we're introspecting the logged values, but it's possible to write a NullOutputStream that doesn't bother to construct the array internally.  (Here's a [Java version](https://github.com/apache/commons-io/blob/master/src/main/java/org/apache/commons/io/output/NullOutputStream.java)).  If a test logs a lot, this will create a large array to be thrown away.  It's probably not a problem in practice, but might be a nice cleanup.

Reply:
done